consumption.tex

```diff
@@ -0,0 +1,62 @@
+
+In the interest of keeping this implementation generic we assume that the user receives a token in return for his burn. However, instead of minting a token, the target cryptocurrency could allow the burn verifier contract to mint native cryptocurrency for any user who successfully claims an event. This would allow the target cryptocurrency to be bootstrapped entirely though burning as desired.
+
+\subsection{Verifying Block Connection}\label{section:block-connection}
+We now shift our attention to the problem of verifying whether a block really belongs in the best source chain. We provide multiple ways of implementing the aforementioned \textsf{is-in-best-chain} method.
+
+\noindent
+\textbf{NIPoPoWs.}
+Verifying block connection can be achieved through NIPoPoWs, as in~\cite{pow-sidechains}.
+We remark that with this setup a block connection proof may be considered valid provisionally, but there needs to be a period in which the proof can be disputed for the smart contract to be certain for the validity of the proof. Specifically, when a user performs a claim they have to put down some collateral. If they have provided a valid NIPoPoW, \textsf{is-in-best-chain} will initially be true. However a challenger can dispute the provided proof which would turn the result of \textsf{is-in-best-chain} to false, abort the claim and grant the challenger the user's collateral. If the contestation period ends with the proof undisputed then the collateral gets returned to the user and the claim is performed successfully.
+
+\noindent
+\textbf{Federation.}
+A simpler approach is to allow a federation of nodes monitoring the source chain to vote for the blocks that are included in the best source chain.
+
+\import{./}{algorithms/alg.is-in-best-chain-federation.tex}
+
+Let us assume that the smart contract holds in its state the continuously updated set of all valid blocks on the best source blockchain, in the form of a Merkle Tree root $\mathcal{M}$. Then, by verifying a Merkle Tree inclusion proof \textsf{blockConnection} for $\textsf{b} \in \mathcal{M}$ a verifier can be certain that the block provided is indeed part of the best source blockchain. Combined with verifying the transaction this completes our {\sf is-in-best-chain}$_\mathcal{M}$ function, shown in Algorithm~\ref{alg.is-in-best-chain-federation}.
```

Comment:
Not sure why there's a need to verify a transaction at this point – we are only verifying block inclusion.

Reply:
oops, meant to remove that but missed it